types/resource.go

```diff
@@ -198,10 +198,23 @@ const (
 	NodeStateDown = NodeState("down")
 )
 
+type NodeConditionType string
+
+const (
+	NodeConditionTypeMountPropagation = "MountPropagation"
+	NodeConditionTypeNodeState        = "NodeState"
+	NodeConditionTypeDiskState        = "DiskState"
+)
+
+const (
+	NodeConditionReasonNotSupportMountPropagation = "NotSupportMountPropagation"
+)
+
 type NodeStatus struct {
-	State            NodeState             `json:"state"`
-	DiskStatus       map[string]DiskStatus `json:"diskStatus"`
-	MountPropagation bool                  `json:"mountPropagation"`
+	DiskStatus map[string]DiskStatus `json:"diskStatus"`
+
+	NodeConditions map[NodeConditionType]Condition `json:"nodeConditions"`
+	DiskConditions map[string]Condition            `json:"diskConditions"`
```

Comment:
Disk conditions should be a map inside DiskStatus. It can be more than one conditions for each disk.